python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -459,30 +440,120 @@ def __init__(self, n=1, normalize=False, **kwds):
             "minute",
             "second",
             "microsecond",
-            "millisecond" "nanosecond",
+            "millisecond",
+            "nanosecond",
+        }
+
+        supported_units = {
+            "years",
+            "months",
+            "weeks",
+            "days",
+            "hours",
+            "minutes",
+            "seconds",
+            "milliseconds",
+            "microseconds",
+            "nanoseconds",
         }
 
-        supported_kwargs = {"months"}
+        unsupported_units = all_possible_units - supported_units
+
+        invalid_kwds = set(kwds) - supported_units - unsupported_units
+        if invalid_kwds:
+            raise TypeError(
+                f"Keyword arguments '{','.join(list(invalid_kwds))}'"
+                " are not recognized"
+            )
+
+        unsupported_kwds = set(kwds) & unsupported_units
+        if unsupported_kwds:
+            raise NotImplementedError(
+                f"Keyword arguments '{','.join(list(unsupported_kwds))}'"
+                " are not yet supported."
+            )
+
+        if any(val != int(val) for val in kwds.values()):
```

Comment:
Do we want to more nicely / efficiently handle the case where `val` is a `cudf.Scalar` of an integer type?

Reply:
Good catch - I added an `is_integer` utility which should be pretty fast for checking if a `cudf.Scalar` is an integer. For all other objects, it falls back to `pd.api.types.is_integer`.